internal/controllers/module_nmc_reconciler.go

```diff
@@ -236,17 +236,17 @@ func (mnrh *moduleNMCReconcilerHelper) enableModuleOnNode(ctx context.Context, m
 	}
 
 	nmc := &kmmv1beta1.NodeModulesConfig{
-		ObjectMeta: metav1.ObjectMeta{Name: nodeName},
+		ObjectMeta: metav1.ObjectMeta{Name: node.Name},
 	}
 
 	opRes, err := controllerutil.CreateOrPatch(ctx, mnrh.client, nmc, func() error {
-		return mnrh.nmcHelper.SetModuleConfig(nmc, mld, &moduleConfig)
+		return mnrh.nmcHelper.SetModuleConfig(nmc, mld, &moduleConfig, node)
```

Comment:
I think I would rather call `controllerutil.SetOwnerReference` here. It does not belong to `SetModuleConfig` in my opinion, which should just manipulate the `spec.modules` list.

Reply:
Fixed